module/reward/reward.py

```diff
@@ -338,45 +343,31 @@ def _reward_mail_selected(self, button, image):
         im = rgb2gray(crop(image, check_area))
         return True if np.mean(im) < 182 else False
 
-    def _reward_mail_collect(self, skip_first_screenshot=True):
+    def _reward_mail_collect_one(self, item, skip_first_screenshot=True):
         """
-        Executes the mail collect sequence
+        Executes the collecting sequence on
+        one mail item
 
         Args:
+            item (Item):
             skip_first_screenshot (bool):
-
-        Returns:
-            bool
         """
-        collectable = self._reward_mail_get_collectable()
-        if not collectable:
-            logger.info('No mail to collect')
-            return False
-
+        btn = item._button
         click_timer = Timer(1.5, count=3)
         while 1:
             if skip_first_screenshot:
                 skip_first_screenshot = False
             else:
                 self.device.screenshot()
 
-            # End
-            if not collectable:
-                break
-            else:
-                btn = collectable[-1]._button
-                if not self._reward_mail_selected(btn, self.device.image) and \
-                   click_timer.reached():
+            if click_timer.reached():
+                if not self._reward_mail_selected(btn, self.device.image):
                     self.device.click(btn.move((350, 0)))
                     self.device.sleep((0.3, 0.5))
                     click_timer.reset()
                     continue
-
-            if self.appear_then_click(MAIL_COLLECT, offset=(20, 20), interval=3):
                 click_timer.reset()
-                continue
-            if self.appear(MAIL_COLLECTED, offset=(20, 20), interval=3):
-                collectable.pop()
+            if self.appear_then_click(MAIL_COLLECT, offset=(20, 20), interval=3):
                 click_timer.reset()
                 continue
             if self.appear_then_click(GET_ITEMS_1, offset=(30, 30), interval=1):
```

Comment:
Click a safe area to avoid exiting the mail page

Reply:
Added `MAIL_CLICK_SAFE_AREA` for the `GET_ITEMS_*` clauses

Reply:
Reuse `MAIL_COLLECT` as `MAIL_CLICK_SAFE_AREA`

Reply:
Didn't notice that, good idea. Deleted `MAIL_CLICK_SAFE_AREA`